android/capacitor/src/main/java/com/getcapacitor/plugin/util/HttpRequestHandler.java

```diff
@@ -238,11 +238,14 @@ public static Object readData(ICapacitorHttpUrlConnection connection, ResponseTy
             } else {
                 return readStreamAsString(errorStream);
             }
-        } else if (contentType != null && contentType.contains(MimeType.APPLICATION_JSON.getValue())) {
-            // backward compatibility
-            return parseJSON(readStreamAsString(connection.getInputStream()));
         } else {
             InputStream stream = connection.getInputStream();
+
```

Comment:
Where I told you to call getInputStream once was a different file where it was being called twice, once for getting the mime type and another one for returning the input stream, and that was not fixed.
Here was not needed because they were different elses, so only one would be called.